ui-validation/src/main/kotlin/edu/stanford/spezi/ui/validation/ReceiveValidation.kt

```diff
@@ -4,19 +4,31 @@ import androidx.compose.runtime.Composable
 import androidx.compose.runtime.CompositionLocalProvider
 import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.MutableState
+import edu.stanford.spezi.ui.validation.internal.LocalCapturedValidationStates
 
 @Composable
 fun ReceiveValidation(
     state: MutableState<ValidationContext>,
     content: @Composable () -> Unit,
+) {
+    ReceiveValidation(
+        onChange = { state.value = it },
+        content = content
+    )
+}
+
+@Composable
+fun ReceiveValidation(
+    onChange: (ValidationContext) -> Unit,
+    content: @Composable () -> Unit,
 ) {
     // This is not remembered on purpose, since we are re-evaluating the validation here.
-    val entries = CapturedValidationStateEntries()
-    CompositionLocalProvider(LocalCapturedValidationStateEntries provides entries) {
+    val entries = mutableListOf<CapturedValidationState>()
```

Comment:
this will be reset to empty in case every recomposition, we should use remembered mutable state instead

Reply:
Yes, this is on purpose. The individual entries are remembered, but we need to recompute this list on every run.